test/puppetlabs/ssl_utils/testutils.clj

```diff
@@ -85,4 +91,91 @@
 (defn str->bytes
   "Turn a `String s` into `bytes[]`"
   [^String s]
-  (bytes (byte-array (map (comp byte int) s))))
\ No newline at end of file
+  (bytes (byte-array (map (comp byte int) s))))
+
+(defn generate-expired-crl
+  [issuer issuer-private-key issuer-public-key]
+  (SSLUtils/generateCRL issuer issuer-private-key issuer-public-key (generate-past-date)))
+
+(defn generate-ca-cert
+  [issuer-name issuer-key-pair serial root?]
+  (let [subject-name (if root?
+                       issuer-name
+                       (cn (format "Intermediate CA %d" serial)))
+        key-pair (if root?
+                  issuer-key-pair
+                  (generate-key-pair 2048))
+        public-key (get-public-key key-pair)
+        not-before (generate-not-before-date)
+        not-after (generate-not-after-date)
+        cert (sign-certificate issuer-name (get-private-key issuer-key-pair)
+                               serial not-before not-after subject-name public-key
+                               (create-ca-extensions subject-name serial public-key))]
+    [cert key-pair]))
+
+(defn generate-cert-chain-with-crls
+  ([number-of-certs]
+   (generate-cert-chain-with-crls number-of-certs generate-crl))
```

Comment:
where is `generate-crl` defined?

Reply:
https://github.com/puppetlabs/jvm-ssl-utils/blob/main/src/clojure/puppetlabs/ssl_utils/core.clj#L492

Reply:
Ooooh, it's coming in from the `:refer :all`, took me a minute to see that.